src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs

```diff
@@ -44,35 +44,28 @@ public async Task AddSemanticClassificationsAsync(Document document, TextSpan te
                 return;
             }
 
-            var remoteSuccess = await TryAddSemanticClassificationsInRemoteProcessAsync(
-                document, textSpan, result, cancellationToken).ConfigureAwait(false);
-            if (remoteSuccess)
-                return;
-
-            using var _ = ArrayBuilder<ClassifiedSpan>.GetInstance(out var temp);
-            await AddSemanticClassificationsInCurrentProcessAsync(
-                document, textSpan, temp, cancellationToken).ConfigureAwait(false);
-            AddRange(temp, result);
-        }
-
-        /// <returns><see langword="true"/> if the remote call was made successfully and we should
-        /// use the results of it. Otherwise, fall back to processing locally</returns>
-        private static async Task<bool> TryAddSemanticClassificationsInRemoteProcessAsync(Document document, TextSpan textSpan, List<ClassifiedSpan> result, CancellationToken cancellationToken)
-        {
             var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
-            if (client == null)
-                return false;
-
-            var classifiedSpans = await client.RunRemoteAsync<SerializableClassifiedSpans>(
-                WellKnownServiceHubService.CodeAnalysis,
-                nameof(IRemoteSemanticClassificationService.GetSemanticClassificationsAsync),
-                document.Project.Solution,
-                new object[] { document.Id, textSpan },
-                callbackTarget: null,
-                cancellationToken).ConfigureAwait(false);
-
-            classifiedSpans.Rehydrate(result);
-            return true;
+            if (client != null)
+            {
+                var classifiedSpans = await client.TryInvokeAsync<IRemoteSemanticClassificationService, SerializableClassifiedSpans>(
+                   document.Project.Solution,
+                   (service, solutionInfo, cancellationToken) => service.GetSemanticClassificationsAsync(solutionInfo, document.Id, textSpan, cancellationToken),
+                   callbackTarget: null,
+                   cancellationToken).ConfigureAwait(false);
+
+                // if the remote call fails do nothing (error has already been reported)
+                if (classifiedSpans.HasValue)
```

Comment:
oh.  i think i see.  it's not a nullable, it's a special Optional type you have to check the value from the invoke itself.  and then the value inside points to the actual result type of hte api.